kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -304,12 +541,30 @@
 
   /**
    * Delete a {@link Dataset} identified by the given dataset URI.
+  * <p>
+   * When you call this method using a dataset URI, both data and metadata are
+   * deleted. After you call this method, the dataset no longer exists, unless
+   * an exception is thrown.
+   * <p>
+   * When you call this method using a view URI, data in that view is deleted.
+   * The dataset's metadata is not changed. This can throw an
```

Comment:
This should state that the dataset's metadata is not deleted, and partition metadata is updated to reflect deleted data.